base/applications/mspaint/winproc.cpp

```diff
@@ -580,6 +582,120 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+typedef ULONG (WINAPI *FN_MAPISendMailW)(LHANDLE, ULONG_PTR, lpMapiMessageW, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszPathName)
+{
+    CStringW strFileTitle;
+    if (PathFileExistsW(pszPathName) && imageModel.IsImageSaved())
+    {
+        strFileTitle = PathFindFileNameW(pszPathName);
+    }
+    else // Not existing
+    {
+        // Delete the temporary file if any
+        if (g_szTempFile[0])
+        {
+            ::DeleteFileW(g_szTempFile);
+            g_szTempFile[0] = UNICODE_NULL;
+        }
+
+        // Get the name of a temporary file
+        WCHAR szTempDir[MAX_PATH];
+        ::GetTempPathW(_countof(szTempDir), szTempDir);
+        if (!::GetTempFileNameW(szTempDir, L"afx", 0, g_szTempFile))
+            return MAPI_E_FAILURE;
+
+        // Get the file type if already saved
+        const GUID *FileType = &GUID_NULL;
+        if (imageModel.IsImageSaved())
+        {
+            CImageDx img;
+            FileType = img.FileTypeFromExtension(PathFindExtensionW(g_szFileName));
```

Comment:
It is weird to create a CImageDx object just to do some parsing with the function.
It would be better if, for the CImageDx class, the FileTypeFromExtension was some kind of static function, so that one could just do instead: `CImageDx::FileTypeFromExtension(etc.);`

Reply:
Fixed.